### PR TITLE
Open folder for previous game history on non-Windows plaforms

### PR DESCRIPTION
**Pull request by EthanArmbrust**
Clicking on "Previous Game History" in "Open" menu opens the folder in Finder on macOS or the default file manager for Linux based systems. Falls back to showing the textbox with the path if this fails.

### CODE REVIEW

File: randovania/gui/main_window.py [hunk 1]
```diff
@@ -539,9 +540,14 @@ def _open_auto_tracker(self):
...
     def _on_menu_action_previously_generated_games(self):
         path = self._options.data_dir.joinpath("game_history")
-        if platform.system() == "Windows":
-            os.startfile(path)
-        else:
+        try:
+            if platform.system() == "Windows":
+                os.startfile(path)
+            elif platform.system() == "Darwin":
+                subprocess.Popen(["open"], path)
```
Comment (henriquegemignani) [inline randovania/gui/main_window.py:547]: This is likely supposed to be `["open", path]`, no?

**Comment by henriquegemignani** [comment]
This is missing some test improvements as well, which is why the tests failed.
Maybe use `subprocess.run` instead of `subprocess.Popen`? Not sure in this precise case.

File: randovania/gui/main_window.py (hunk 1 above)
Comment (EthanArmbrust) [reply randovania/gui/main_window.py:547]: Yep. I'm going to try to test this on an old macbook to make sure the macOS stuff works, as well as make sure the test covers macOS

**Comment by EthanArmbrust** [comment]
I've tested this works on macOS, and tests have been updated to include macOS and OSErrors

**Pull request merged**